CRM/Member/Form.php

```diff
@@ -77,6 +77,23 @@ class CRM_Member_Form extends CRM_Contribute_Form_AbstractEditPayment {
   public $_priceSetId;
 
   /**
+   * Used when price set specified.  Used to indicate that form is only to retrieve
+   * price set.  How we use price set is unconventional.  We re-use the same
+   * Membership/MembershipRenewal Form used to enter memberships.  When user selects
+   * a PriceSet, we use JavaScript to inject HTML returned by a recursive call to this form.
+   *
+   * This flag controls this recursive call, short-circuiting it to only retrieve the price
```

Comment:
That is fair enough.  My main concern would be that it never gets merged, and that this client be "stuck" with a PR that they have to keep rebasing & pulling locally on every upgrade.  But that doesn't sound like it is the case.

Although I'm out of dev cycles for the next month or so, I will certainly provide in the short term any "bug" support on this.  In the new year, you can count on me for #1 unit tests and your PR #9387 (extracting the price set into its own form).